File: src/dataset/python/triangle_exercism-ThomasZumsteg/claude/ClaudeSonnet4_triangle_v2.py
"""Validate and specify triangle types"""

class TriangleError(Exception):
    """Error if it cannot be a triangle"""
    pass

class Triangle:
    """Validate and specify triangle types"""
    __slots__ = ('a', 'b', 'c')
    
    def __init__(self, a, b, c):
        """Create and validate triangle"""
        if a > b:
            if a > c:
                if b > c:
                    self.a, self.b, self.c = c, b, a
                else:
                    self.a, self.b, self.c = b, c, a
            else:
                self.a, self.b, self.c = b, a, c
        else:
            if b > c:
                if a > c:
                    self.a, self.b, self.c = c, a, b
                else:
                    self.a, self.b, self.c = a, c, b
            else:
                self.a, self.b, self.c = a, b, c
        
        if self.a + self.b <= self.c:
            raise TriangleError("Not a Triangle")

    def kind(self):
        """Specify triangle type"""
        if self.a == self.c:
            return "equilateral"
        elif self.a == self.b or self.b == self.c:
            return "isosceles"
        else:
            return "scalene"

File: src/dataset/javascript/triangle_exercism-elenamountz/claude/ClaudeSonnet4_triangle_v2.js
export default class Triangle{
  constructor(s1, s2, s3){
      this.s1 = s1;
      this.s2 = s2;
      this.s3 = s3;
  }
  kind(){
    const s1 = this.s1;
    const s2 = this.s2;
    const s3 = this.s3;
    
    if(s1 <= 0 || s2 <= 0 || s3 <= 0){
      throw new Error;
    }
    if(s1 + s2 <= s3 || s2 + s3 <= s1 || s3 + s1 <= s2){
      throw new Error;
    }
    
    const s1EqS2 = s1 === s2;
    const s1EqS3 = s1 === s3;
    const s2EqS3 = s2 === s3;
    
    if(s1EqS2 && s1EqS3){
      return 'equilateral';
    } else if(s1EqS2 || s1EqS3 || s2EqS3){
      return 'isosceles';
    } else {
      return 'scalene';
    }
  }
};